src/sources/BlackmagicATEM.ts

```diff
@@ -127,6 +130,9 @@ export class BlackmagicATEMSource extends TallyInput {
     }
 
     private processATEMTally(): void {
+        const areSetsEqual = (a, b) => a.size === b.size && [...a].every(value => b.has(value)); //https://stackoverflow.com/a/44827922
+        if(areSetsEqual(this.prvList, this.oldPrvList) && areSetsEqual(this.pgmList, this.oldPgmList)) return;

```

Comment:
Is this only a problem with the atem source? Maybe it would make sense to do that one level "higher", meaning that TA filters duplicate tally data. Then, we would not have to check for that in each source.